data-layers/data-layer-api-caller/src/main/java/com/flipkart/aesop/apicallerdatalayer/delete/ApiCallerDeleteDataLayer.java

```diff
@@ -0,0 +1,71 @@
+package com.flipkart.aesop.apicallerdatalayer.delete;
+
+import com.flipkart.aesop.destinationoperation.DeleteDestinationStoreProcessor;
+import com.flipkart.aesop.event.AbstractEvent;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.linkedin.databus.client.pub.ConsumerCallbackResult;
+import org.json.JSONObject;
+import org.trpr.platform.core.impl.logging.LogFactory;
+import org.trpr.platform.core.spi.logging.Logger;
+
+import java.io.DataOutputStream;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.Iterator;
+import java.util.Map;
+
+/**
+ * Created by aman.gupta on 12/08/15.
+ */
+public class ApiCallerDeleteDataLayer extends DeleteDestinationStoreProcessor {
+
+    private static final Logger LOGGER = LogFactory.getLogger(ApiCallerDeleteDataLayer.class);
+    private URL url;
+    private JSONObject headers;
+    public ApiCallerDeleteDataLayer(URL url, JSONObject headers) {
+        this.url = url;
+        this.headers = headers;
+    }
+
+    @Override
+    protected ConsumerCallbackResult delete(AbstractEvent event) {
+        try {
+            Map<String,Object> eventMap = event.getFieldMapPair();
+            Gson gson = new GsonBuilder().serializeNulls().create();
+            String param = gson.toJson(eventMap);
+            LOGGER.info("Making a post call to url: "+url+" with payload as: "+param);
+            HttpURLConnection con = (HttpURLConnection)url.openConnection();
```

Comment:
A good idea would be to use java-cas-client-0.0.17.4-auth.jar. This will make sure that your implementation can invoke AuthN & CAS enabled clients comfortably. Also, the custom headers option can be used to add additional headers while making the call. Also look at Varadhi code base. It is very similar to what you do since the varadhi consumer is also a HTTP consumer who reads from a Kafka queue and calls a HTTP endpoint.